### ISSUE TITLE
Add support Kotlin methods with spaces

### ISSUE THREAD

**Issue by paplorinc** [melix/jmh-gradle-plugin#150]
In Kotlin you can create method names with spaces - they're supported in JUnit already.
E.g.
```java
@Benchmark fun `transitive dependency calculation`() = dependencies.transitive(root)
```

The bytecode will actually contain spaces, so it's possible the problem is actually in JMH generators, not your plugin:
```// declaration: java.util.Set<com.lorinc.gradle.Dependency> transitive dependency calculation()
public final transitive dependency calculation()Ljava/util/Set;```

The error I'm getting when using such a method is the following:
```
> A failure occurred while executing me.champeau.gradle.JmhBytecodeGeneratorRunnable
   > Generation of JMH bytecode failed with 1 errors:
       - Group name should be the legal Java identifier.
        [org.openjdk.jmh.generators.reflection.RFMethodInfo@27328ae]
```

If the problem is indeed in JMH, could we provide a workaround from the plugin to e.g. add underlines to the method names before generation or something?

Thanks!

**Comment by melix** [comment]
This is probably an issue with JMH itself. Did you try to upgrade to the latest version?

**Comment by paplorinc** [comment]
You mean compile it from https://hg.openjdk.java.net/code-tools/jmh? No, used the version from the plugin.

**Comment by paplorinc** [comment]
I guess the problem is with https://github.com/msteindorfer/jmh/blob/master/jmh-core/src/main/java/org/openjdk/jmh/generators/core/BenchmarkGeneratorUtils.java#L85

**Comment by melix** [comment]
No, I mean the plugin lets you choose the version of JMH to use.

**Comment by melix** [comment]
Likely, yes.

**Comment by paplorinc** [comment]
https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-4.html#jvms-4.2.1 also suggests it should accept spaces.
